weld-spi/src/main/java/org/jboss/weld/manager/api/ExecutorServices.java

```
@@ -39,6 +40,10 @@
 
     ExecutorService getTaskExecutor();
 
+    default ScheduledExecutorService getTimerExecutor() {
+        return Executors.newScheduledThreadPool(1);
```

The default impl is disputable. Each invocation of this method would result in new pool and the caller would be responsible for shutdown etc. I think we should rather throw unsupported operation exception or something like that.